lib/rules/no-multiple-assertions-wait-for.ts

```diff
@@ -0,0 +1,67 @@
+import { ESLintUtils, TSESTree } from '@typescript-eslint/experimental-utils'
+import { getDocsUrl } from '../utils'
+import { isBlockStatement, findClosestCalleName, isMemberExpression, isCallExpression, isIdentifier } from '../node-utils'
+
+export const RULE_NAME = 'no-multiple-expect-wait-for';
+
+const WAIT_EXPRESSION_QUERY =
+  'CallExpression[callee.name=/^(waitFor)$/]';
+
+export type MessageIds = 'noMultipleAssertionWaitFor';
+type Options = [];
+
+export default ESLintUtils.RuleCreator(getDocsUrl)<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        "It's preferred to avoid multiple assertions in `waitFor`",
+      category: 'Best Practices',
+      recommended: false,
+    },
+    messages: {
+      noMultipleAssertionWaitFor: 'Avoid use multiple assertions to `{{ methodName }}`',
+    },
+    fixable: null,
+    schema: [],
+  },
+  defaultOptions: [],
+  create: function(context) {
+    function reporttMultipleAssertion(
+      node: TSESTree.BlockStatement
+    ) {
+      const hasMultipleExpects = (body: Array<TSESTree.Node>): boolean =>
+        body.every((node: TSESTree.ExpressionStatement) => {
+          if (
+            isCallExpression(node?.expression) &&
+            isMemberExpression(node?.expression?.callee) &&
+            isCallExpression(node?.expression?.callee?.object)
+          ) {
+            const object: TSESTree.CallExpression = node?.expression?.callee?.object
+            const expressionName: string = (object?.callee as TSESTree.Identifier)?.name
+            return expressionName === 'expect'
+          } else {
+            return false
+          }
+        })
```

Comment:
I think here instead of confirming _every_ statement to be a `expect`, we should check filter and check the length is large or equals than two. Otherwise, a `console.log` in-between will make this `every` return `false`. For instance, I think scenario

```ts
waitFor(() => {
  expect(a).toEqual('a')
  console.log('asserting b')
  expect(b).toEqual('b')
})
```

will not be detected by this code